File: 9-palindrome-number/palindrome-number.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x < 0 or (x != 0 and (x % 10) == 0):
            return False

        reverse = 0
        copy = x

        while copy:
            rem = copy % 10
            reverse = reverse * 10 + rem
            copy = copy // 10

        return True if x == reverse else False

File: 9-palindrome-number/palindrome-number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        if(x < 0 || (x > 0 && x % 10 == 0)) return false;

        long long copy = x;
        long long reverse = 0;

        while(copy){
            reverse = reverse * 10 + (copy % 10);
            copy = copy / 10;
        }

        return x == reverse;
    }
};